File: js_project/client/src/epubfile/html/TextNode.js
import BaseNode from './BaseNode'
// 文本类型节点 分纯文本和图片文字(由客户端处理)
export default class TextNode extends BaseNode {
    textType // 文本类型 plain image
    content // 文本内容
    constructor(props) {
        super(props)
        this.nodeType = 'text'
        this.textType = 'plain'
        this.content = props && props.content
    }
}

File: lib/src/main/java/com/chineseall/epubparser/lib/html/TextNode.kt
package com.chineseall.epubparser.lib.html

import com.chineseall.epubparser.lib.render.RenderContext
import com.chineseall.epubparser.lib.render.RenderSection
import com.chineseall.epubparser.lib.render.TextSection

// 文本节点
class TextNode(
    nodeType: String?,
    type: String?,
    className: String?,
    style: String?,
    bookPlot: BookPlot?,
    var textType: String?,
    var content: String?
) : BaseNode(nodeType, type, className, style, bookPlot) {
    override fun appendSection(renderContext: RenderContext, sections: MutableList<RenderSection>) {
        if (newSection() || sections.isEmpty()) {
            // 新建
            val textSection = TextSection()
            textSection.appendNode(this)
            sections.add(textSection)
        } else {
            // 追加
            val tailSection = sections[0]
            if (tailSection is TextSection) {
                tailSection.appendNode(this)
            }
        }
    }
}